src/Selenium2Library/keywords/_logging.py

```diff
@@ -12,7 +12,17 @@ def _debug(self, message):
         logger.debug(message)
 
     def _get_log_dir(self):
-        variables = BuiltIn().get_variables()
+        # Use screenshot root directory if set
+        if self.screenshot_root_directory is not None:
+            return self.screenshot_root_directory
+
+        # If robotframework isn't running use current directory
+        try:
+            variables = BuiltIn().get_variables()
+        except RobotNotRunningError:
+            return os.getcwd()
```

Comment:
1. Whoops, nice catch. I haven't written a test for this yet, but I certainly will before its merged in. 
2. Simply because I like to be explicit. 
3. If we're going to add it, we may as well make it available to everyone. I actually added it as an argument into the S2L constructor so it'd be a newly documented feature.